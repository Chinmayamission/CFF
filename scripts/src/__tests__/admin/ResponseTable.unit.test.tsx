import { mount } from 'enzyme';
import React from "react";
import { Provider } from "react-redux";
import ResponseTable from "../../admin/ResponseTable/ResponseTable";
import store from "../../store";

it('renders default response table', () => {
  // let response: IResponse = { "payment_status_detail": [], "value": {}, "amount_paid": "0", "paid": false, "payment_trail": null };
  // store.dispatch(setResponseDetail(response))
  const wrapper = mount(
    <Provider store={store}><ResponseTable match={{ params: { formId: 123 } }} /></Provider>
  );
  expect(wrapper).toMatchSnapshot();
});

// store.dispatch(setRenderedForm({"schema": schema, "uiSchema": {}, "name": "Form", "formOptions": {"paymentInfo": null, "paymentMethods": null, "confirmationEmailInfo": null, "dataOptions": dataOptions} } ));
// store.dispatch(setResponses([]));
// store.dispatch(setResponsesSelectedView("main"));
// const wrapper = render(
//   <Provider store={store}><ResponseTableView /></Provider>
// );
// expect(wrapper).toMatchSnapshot();