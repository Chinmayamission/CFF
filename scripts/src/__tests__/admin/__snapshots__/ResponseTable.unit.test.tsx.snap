// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`responses with default data 1`] = `
<div>
  <ul
    class="nav nav-pills"
  >
    <li
      class="nav-item btn-outline-primary"
    >
      <a
        class="nav-link"
      >
        All Responses
      </a>
    </li>
    <li
      class="nav-item btn-outline-primary"
    >
      <a
        class="nav-link"
      >
        Adults Display Name
      </a>
    </li>
    <li
      class="nav-item btn-outline-primary"
    >
      <a
        class="nav-link"
      >
        Children Display Name
      </a>
    </li>
  </ul>
  <button
    class="btn btn-outline-primary"
  >
    Download CSV
  </button>
  <div
    class="ReactTable"
  >
    <div
      class="rt-table"
      role="grid"
    >
      <div
        class="rt-thead -header"
        style="min-width:835px"
      >
        <div
          class="rt-tr"
          role="row"
        >
          <div
            class="rt-th"
            role="columnheader"
            style="flex:35 0 auto;width:35px;max-width:35px"
            tabindex="-1"
          >
            <div
              class=""
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Parent Names
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Phone numbers
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Parent email addresses
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Home address
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Child first names
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Returning family
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Fees due
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Amount paid
            </div>
            <div
              class="rt-resizer"
            />
          </div>
        </div>
      </div>
      <div
        class="rt-thead -filters"
        style="min-width:835px"
      >
        <div
          class="rt-tr"
          role="row"
        >
          <div
            class="rt-th"
            role="columnheader"
            style="flex:35 0 auto;width:35px;max-width:35px"
            tabindex="-1"
          />
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="rt-tbody"
        style="min-width:835px"
      >
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -odd"
            role="row"
            style="color:black"
          >
            <div
              class="rt-td  rt-expandable"
              role="gridcell"
              style="flex:35 0 auto;width:35px;max-width:35px"
            >
              <div
                class="rt-expander"
              >
                â€¢
              </div>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              mom ram, dad ram
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              1231234311 1231231233, 1231231233
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              mom@chinmayamission.com, dad@chinmayamission.com
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              123 abc st  johns creek GA 30022
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              ash, ash2
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            />
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              $510.00
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              $100.00
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pagination-bottom"
    >
      <div
        class="-pagination"
      >
        <div
          class="-previous"
        >
          <button
            class="-btn"
            disabled=""
            type="button"
          >
            Previous
          </button>
        </div>
        <div
          class="-center"
        >
          <span
            class="-pageInfo"
          >
            Page 
            <div
              class="-pageJump"
            >
              <input
                type="number"
                value="1"
              />
            </div>
             of 
            <span
              class="-totalPages"
            >
              1
            </span>
          </span>
          <span
            class="select-wrap -pageSizeOptions"
          >
            <select>
              <option
                value="5"
              >
                5 rows
              </option>
              <option
                value="10"
              >
                10 rows
              </option>
              <option
                selected=""
                value="20"
              >
                20 rows
              </option>
              <option
                value="25"
              >
                25 rows
              </option>
              <option
                value="50"
              >
                50 rows
              </option>
              <option
                value="100"
              >
                100 rows
              </option>
            </select>
          </span>
        </div>
        <div
          class="-next"
        >
          <button
            class="-btn"
            disabled=""
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
    <div
      class="-loading"
    >
      <div
        class="-loading-inner"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;

exports[`responses with unwind data 1`] = `
<div>
  <ul
    class="nav nav-pills"
  >
    <li
      class="nav-item btn-outline-primary"
    >
      <a
        class="nav-link"
      >
        All Responses
      </a>
    </li>
    <li
      class="nav-item btn-outline-primary"
    >
      <a
        class="nav-link"
      >
        Adults Display Name
      </a>
    </li>
    <li
      class="nav-item btn-outline-primary"
    >
      <a
        class="nav-link"
      >
        Children Display Name
      </a>
    </li>
  </ul>
  <button
    class="btn btn-outline-primary"
  >
    Download CSV
  </button>
  <div
    class="ReactTable"
  >
    <div
      class="rt-table"
      role="grid"
    >
      <div
        class="rt-thead -header"
        style="min-width:1235px"
      >
        <div
          class="rt-tr"
          role="row"
        >
          <div
            class="rt-th"
            role="columnheader"
            style="flex:35 0 auto;width:35px;max-width:35px"
            tabindex="-1"
          >
            <div
              class=""
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              First Name
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Last Name
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Gender
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Date of Birth
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Grade
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Email
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Allergies
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Parent Names
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Phone numbers
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Parent email addresses
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Home address
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th  rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Volunteering
            </div>
            <div
              class="rt-resizer"
            />
          </div>
        </div>
      </div>
      <div
        class="rt-thead -filters"
        style="min-width:1235px"
      >
        <div
          class="rt-tr"
          role="row"
        >
          <div
            class="rt-th"
            role="columnheader"
            style="flex:35 0 auto;width:35px;max-width:35px"
            tabindex="-1"
          />
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
          <div
            class="rt-th"
            role="columnheader"
            style="flex:100 0 auto;width:100px"
            tabindex="-1"
          >
            <input
              style="width:100%"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="rt-tbody"
        style="min-width:1235px"
      >
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -odd"
            role="row"
            style="color:black"
          >
            <div
              class="rt-td  rt-expandable"
              role="gridcell"
              style="flex:35 0 auto;width:35px;max-width:35px"
            >
              <div
                class="rt-expander"
              >
                â€¢
              </div>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              ash2
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              ram
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              M
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              9301-01-14
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              Toddler
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              sib@chinmayamission.com
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              onions and garlic
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              mom ram, dad ram
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              1231234311 1231231233, 1231231233
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              mom@chinmayamission.com, dad@chinmayamission.com
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              123 abc st  johns creek GA 30022
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              Art activity Dance, Teaching Dance Technology Cooking Event organization
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -even"
            role="row"
            style="color:black"
          >
            <div
              class="rt-td  rt-expandable"
              role="gridcell"
              style="flex:35 0 auto;width:35px;max-width:35px"
            >
              <div
                class="rt-expander"
              >
                â€¢
              </div>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              ash
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              ram
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              M
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              1996-07-12
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              Pre K
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            />
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            />
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              mom ram, dad ram
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              1231234311 1231231233, 1231231233
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              mom@chinmayamission.com, dad@chinmayamission.com
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              123 abc st  johns creek GA 30022
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex:100 0 auto;width:100px"
            >
              Art activity Dance, Teaching Dance Technology Cooking Event organization
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pagination-bottom"
    >
      <div
        class="-pagination"
      >
        <div
          class="-previous"
        >
          <button
            class="-btn"
            disabled=""
            type="button"
          >
            Previous
          </button>
        </div>
        <div
          class="-center"
        >
          <span
            class="-pageInfo"
          >
            Page 
            <div
              class="-pageJump"
            >
              <input
                type="number"
                value="1"
              />
            </div>
             of 
            <span
              class="-totalPages"
            >
              1
            </span>
          </span>
          <span
            class="select-wrap -pageSizeOptions"
          >
            <select>
              <option
                value="5"
              >
                5 rows
              </option>
              <option
                value="10"
              >
                10 rows
              </option>
              <option
                selected=""
                value="20"
              >
                20 rows
              </option>
              <option
                value="25"
              >
                25 rows
              </option>
              <option
                value="50"
              >
                50 rows
              </option>
              <option
                value="100"
              >
                100 rows
              </option>
            </select>
          </span>
        </div>
        <div
          class="-next"
        >
          <button
            class="-btn"
            disabled=""
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
    <div
      class="-loading"
    >
      <div
        class="-loading-inner"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;
